consumers/campaigner/main.go

```diff
@@ -66,128 +102,134 @@ func (h *MessageHandler) HandleMessage(m *nsq.Message) error {
 			logrus.WithFields(logrus.Fields{
 				"user_id":     msg.UserID,
 				"segment_ids": msg.SegmentIDs,
-			}).WithError(err).Error("Unable to fetch subscribers.")
-			break
+			}).WithError(err).Error("unable to fetch subscribers.")
+			return nil
 		}
 
 		if len(subs) == 0 {
 			break
 		}
 
-		// SES allows to send 50 emails in a bulk sending operation
-		chunkSize := 50
-		for i := 0; i < len(subs); i += chunkSize {
-			end := i + chunkSize
-			if end > len(subs) {
-				end = len(subs)
-			}
-
-			var dest []*ses.BulkEmailDestination
-			for _, s := range subs[i:end] {
-				m, err := s.GetMetadata()
-				if err != nil {
-					logrus.WithError(err).
-						WithField("subscriber", s).
-						Error("Unable to get subscriber metadata.")
-
-					continue
-				}
-
-				if s.Name != "" {
-					m["name"] = s.Name
-				}
-
-				url, err := s.GetUnsubscribeURL(msg.UserUUID)
-				if err != nil {
-					logrus.WithError(err).
-						WithField("subscriber", s).
-						Error("Unable to get unsubscribe url.")
-				} else {
-					m["unsubscribe_url"] = url
-				}
-
-				jsonMeta, err := json.Marshal(m)
-				if err != nil {
-					logrus.WithError(err).
-						WithField("subscriber", s).
-						Error("Unable to marshal metadata to json.")
-
-					continue
-				}
-				d := &ses.BulkEmailDestination{
-					Destination: &ses.Destination{
-						ToAddresses: []*string{aws.String(s.Email)},
-					},
-					ReplacementTemplateData: aws.String(string(jsonMeta)),
-				}
-
-				dest = append(dest, d)
-			}
+		template, err := h.svc.GetTemplate(context.Background(), msg.TemplateID, msg.UserID)
+		if err != nil {
+			logrus.WithError(err).
+				WithFields(logrus.Fields{
+					"template_id": msg.TemplateID,
+					"user_id":     msg.UserID,
+				}).
+				Error("unable to get template")
+			return nil
+		}
+		html, err := mustache.ParseString(template.HTMLPart)
+		// todo retry = true/false?
+		if err != nil {
+			logrus.WithError(err).
+				WithField("html_part", template.HTMLPart).
```

Comment:
No need to add the whole html part, add IDs such as: `campaign_id`, `user_id`, `template_id` etc.